src/default.js

```diff
@@ -103,34 +94,34 @@ OC = new Vue({
 						timeout: 0
 					});
 				}
-
 			});
 		},
 
 
 		/**
 		 * Setup all available apps
-		 *
-		 * @return Promise
 		 */
 
 		_setupApps () {
 
-			_.forEach(this.apps, (app, i) => {
+			_.forEach(this.apps, (app, index) => {
 
 				// TODO: Find better var name for 'foo'
-				requirejs([this.appJS(app.id, 'boot')], ( foo ) => {
-					let defaults = {
-						enabled : true,
-						running : false,
-					};
-
-					this.apps[i] = _.assignIn(defaults, foo.info);
-
-					// inject self
-					foo.setup(foo).then(() => {
-						if (this.apps.length === ++i) {
-							this.$bus.emit('phoenix:apps-setup')
+				requirejs([this.appJS(app.id, 'boot')], ( a ) => {
+
+					a.setup().then((appInfo) => {
+						if (app.id != appInfo.id) {
```

Comment:
use strict equality

in general if you use a linter like jslint, jshint or eslint it should tell you about this